faq.md

```diff
@@ -112,162 +111,135 @@ It’s a component that was introduced recently. Essentially it’s a stateless
-
-One option is to use `publishTimestamp` of messages. When you start replaying you can check current "now", then you replay util you hit a message with timestamp >= now.
+There is no direct way because messages can still be published in the topic, and relying on the `readNext(timeout)` is not precise because the client might be temporarily disconnected from broker at that moment.
 
-Another option is to "terminate" the topic. Once a topic is "terminated", no more message can be published on the topic, a reader/consumer can check the `hasReachedEndOfTopic()` condition to know when that happened.
+One option is to use `publishTimestamp` of messages. When you start replaying, you can check current "now", and then you replay until you hit a message with timestamp >= now.
 
-A final option is to check the topic stats. This is a tiny bit involved, because it requires the admin client (or using REST) to get the stats for the topic and checking the "backlog". If the backlog is 0, it means we've hit the end.
+Another option is to "terminate" the topic. Once a topic is "terminated", no more message can be published on the topic. To know when a topic is terminated, you can check the `hasReachedEndOfTopic()` condition.
 
-### How can I prevent an inactive topic to be deleted under any circumstance? I want to set no time or space limit for a certain namespace.
-There’s not currently an option for “infinite” (though it sounds a good idea! maybe we could use `-1` for that). The only option now is to use INT_MAX for `retentionTimeInMinutes` and LONG_MAX for `retentionSizeInMB`. It’s not “infinite” but 4085 years of retention should probably be enough!
+A third option is to check the topic stats. This is a tiny bit involved, because it requires the admin client (or using REST) to get the stats for the topic and checking the "backlog". If the backlog is 0, it means you've hit the end.
 
-### Is there a profiling option in Pulsar, so that we can breakdown the time costed in every stage? For instance, message A stay in queue 1ms, bk writing time 2ms(interval between sending to bk and receiving ack from bk) and so on.
-There are latency stats at different stages. In the client (eg: reported every 1min in info logs). 
-In the broker: accessible through the broker metrics, and finally in bookies where there are several different latency metrics. 
+### How can I prevent an inactive topic to be deleted under any circumstance? I do not want to set time or space limit for a certain namespace.
+Currently, there is not an option for “infinite” (though it sounds a good idea! Maybe we could use `-1` for that). The only option is to use INT_MAX for `retentionTimeInMinutes` and LONG_MAX for `retentionSizeInMB`. It’s not “infinite”, yet 4085 years of retention is probably be enough!
+<!--@Sijie: could you please refine this answer when you read here?-->
```

Comment:
@Jennifer88huang 

The answer is out of date.

Pulsar supports infinite retention now. See section: http://pulsar.apache.org/docs/en/cookbooks-retention-expiry/#retention-policies